starport/pkg/chaincmd/runner/runner.go

```diff
@@ -127,5 +137,25 @@ func (r Runner) run(ctx context.Context, roptions runOptions, soptions ...step.O
 		New(rnoptions...).
 		Run(ctx, step.New(soptions...))
 
-	return errors.Wrap(err, errb.String())
+	if roptions.wrapStdErr {
+		return errors.Wrap(err, errb.buf.String())
+	}
+	return err
+}
+
+type truncatedBuffer struct {
```

Comment:
Perfect! Can we move this to a seperate package under `pkg/`, make it newable and add a test? It would be ideal to have a default cap that can be overwritten by a functional option.